helm-chart/binderhub/schema.yaml

```diff
@@ -349,14 +350,17 @@ properties:
           See the [Kubernetes docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
           to learn more about labels.
 
+  containerBuilderPod:
+    type: string
+    enum: ["disabled", "dind", "pink"]
+    default: "disabled"
+    description: |
+      Selected builder pod type
+
   dind:
     type: object
     additionalProperties: false
     properties:
-      enabled:
```

Comment:
Is it reasonable to deprecate instead of making a breaking change?

Reply:
Yes, we can do it that way. With a mention in the `NOTES.txt` as it seems that helm does not have yet the support for handling deprecation in the schema.

Reply:
I suggest we go for a hard deprecation, forcing users to transition as part of this upgrade. Since we don't have a nice version control, let's go for a decisive and verbose error that will fail. This failure will happen on template rendering, and doesn't risk breaking anything in the k8s cluster.

1. Keep what is being removed allowed in the schema, otherwise `helm` with error with "not in schema" before you can provide nice deprecation messages.
2. Let NOTES.txt handle the deprecation logic, copy paste from https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/2.0.0/jupyterhub/templates/NOTES.txt#L116-L153 and adjust.

Reply:
The main reason I suggested a deprecation was so this PR could be merged as soon as possible with the decision over the breaking change being made in a follow-up. I'm happy with either option though.

Reply:
I'm super happy about a breaking change in the helm chart's schema if implemented with a clear error message like above, and would prefer it over growing complexity.

This kind of breaking changes has been very smooth to document and migrate historically, lets do it!

Reply:
If I get things correctly, I should remove the warning I added in `NOTES.txt` file and make it error direcly if `.Values.dind.enabled` is found rather than the current check that checks both this and the `containerBuilderPod` value ?

Reply:
Thanks @sgaist, excellent work!!